### PR TITLE
Add cobra for building out the cli

### LINKED ISSUE #7: "k3ai-client" is bit verbose
How about using k3ai as the CLI name? Are there any plans to introduce a server component in the future?

### PR DESCRIPTION
**Pull request by harsimranmaan**
cobra is a popular library for building golang cli. kubectl, qbec and many other
tools in the community use cobra. This PR bootstraps the project so that new commands and flags
can be built with ease.
Closes #7

### CODE REVIEW

**Comment by saiyam1814** [comment]
I like the Idea of cobra for cli.

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"fmt"
+	"os"
+
+	"github.com/kf5i/k3ai-core/pkg/k8s/kctl"
+	"github.com/kf5i/k3ai-core/pkg/plugins"
+	"github.com/spf13/cobra"
+)
+
+const k3aiBinaryName = "k3ai-client"
```
Comment (Gsantomaggio) [inline cmd/root.go:12]: `k3ai-cli` would be better !

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+var rootCmd = &cobra.Command{
+	Use:   k3aiBinaryName,
+	Short: fmt.Sprintf(`%s installs AI tools`, k3aiBinaryName),
+	Long: fmt.Sprintf(` %s is a lightweight infrastructure-in-a-box solution specifically built to
+	install and configure AI tools and platforms in production environments on Edge
+	and IoT devices as easily as local test environments.`, k3aiBinaryName),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		pluginList, _ := plugins.GetPluginList()
```
Comment (Gsantomaggio) [inline cmd/root.go:21]: I would handle the error here

File: cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package cmd
```
Comment (Gsantomaggio) [inline cmd/root.go:1]: Can you change the name, and put it in a subdirectory? in the future we will have also a web part. 

so it should be something like:
 - `cmd\tools\cli\cli.go`
 - `cmd\tools\web\web.go`

File: cmd/root.go (hunk 1 above)
Comment (harsimranmaan) [reply cmd/root.go:12]: yes agreed.

File: cmd/root.go (hunk 2 above)
Comment (harsimranmaan) [reply cmd/root.go:21]: This was copied verbatim from main.go. I'd also like to move commands from the root to something like `k3ai-cli apply`, `k3ai-cli delete` etc. But I don't have enough context yet to make that call.

File: cmd/root.go (hunk 1 above)
Comment (harsimranmaan) [reply cmd/root.go:12]: If you could merge #4 , I'll make the required changes as there are a few places where this needs to be addressed

File: cmd/root.go (hunk 2 above)
Comment (Gsantomaggio) [reply cmd/root.go:21]: >This was copied verbatim from main.go.
You are right, my fault!
> I'd also like to move commands from the root to something like k3ai-cli apply, k3ai-cli delete
You are right, but let's proceed by steps

File: cmd/root.go (hunk 1 above)
Comment (Gsantomaggio) [reply cmd/root.go:12]: >If you could merge #4 

Done!

**Review by Gsantomaggio — APPROVED** [review]

**Comment by Gsantomaggio** [comment]
Thank you @harsimranmaan !

**Pull request merged**